join-statements/joinstatements.md

```diff
@@ -0,0 +1,423 @@
+ Roger   | Thursday    |     0
+ Roger   | Friday      |     0
+ Roger   | Saturday    |     0
+ Roger   | Sunday      |     0
+ Roger   | Saturday    |     1
+ Roger   | Sunday      |     1
+ Roger   | Saturday    |     2
+ Roger   | Sunday      |     2
+ Justine | Monday      |     1
+ Justine | Tuesday     |     1
+ Justine | Wednesday   |     1
+ Justine | Thursday    |     1
+ Justine | Friday      |     1
+ Justine | Monday      |     2
+ Justine | Tuesday     |     2
+ Justine | Wednesday   |     2
+ Justine | Thursday    |     2
+ Justine | Friday      |     2
+ Justine | Monday      |     0
+ Justine | Tuesday     |     0
+ Justine | Wednesday   |     0
+ Justine | Thursday    |     0
+ Justine | Friday      |     0
+ Justine | Saturday    |     0
+ Justine | Sunday      |     0
+ Justine | Saturday    |     1
+ Justine | Sunday      |     1
+ Justine | Saturday    |     2
+ Justine | Sunday      |     2
+ Omar    | Monday      |     1
+ Omar    | Tuesday     |     1
+ Omar    | Wednesday   |     1
+ Omar    | Thursday    |     1
+ Omar    | Friday      |     1
+ Omar    | Monday      |     2
+ Omar    | Tuesday     |     2
+ Omar    | Wednesday   |     2
+ Omar    | Thursday    |     2
+ Omar    | Friday      |     2
+ Omar    | Monday      |     0
+ Omar    | Tuesday     |     0
+ Omar    | Wednesday   |     0
+ Omar    | Thursday    |     0
+ Omar    | Friday      |     0
+ Omar    | Saturday    |     0
+ Omar    | Sunday      |     0
+ Omar    | Saturday    |     1
+ Omar    | Sunday      |     1
+ Omar    | Saturday    |     2
+ Omar    | Sunday      |     2
+ Tammy   | Monday      |     1
+ Tammy   | Tuesday     |     1
+ Tammy   | Wednesday   |     1
+ Tammy   | Thursday    |     1
+ Tammy   | Friday      |     1
+ Tammy   | Monday      |     2
+ Tammy   | Tuesday     |     2
+ Tammy   | Wednesday   |     2
+ Tammy   | Thursday    |     2
+ Tammy   | Friday      |     2
+ Tammy   | Monday      |     0
+ Tammy   | Tuesday     |     0
+ Tammy   | Wednesday   |     0
+ Tammy   | Thursday    |     0
+ Tammy   | Friday      |     0
+ Tammy   | Saturday    |     0
+ Tammy   | Sunday      |     0
+ Tammy   | Saturday    |     1
+ Tammy   | Sunday      |     1
+ Tammy   | Saturday    |     2
+ Tammy   | Sunday      |     2
+(126 rows)
+```
+
+8) 
+
+SELECT v.name, d.name
+FROM volunteers AS v
+LEFT OUTER JOIN dogs AS d
+ON v.foster_id = d.id;
+
+SELECT a.first_name, a.last_name, d.name, c.name
+FROM adoptions AS ad
+LEFT JOIN dogs AS d ON d.id = ad.dog
+LEFT JOIN cats AS c ON c.id = ad.cat
+JOIN adopters AS a ON ad.adopter = a.id
+
+SELECT a.first_name, a.last_name, d.name
+FROM adopters AS a
+LEFT JOIN adoptions AS ad ON a.id = ad.adopter
+CROSS JOIN dogs AS d
+WHERE ad.id IS NULL AND d.adoption_date IS NULL;
+
+
+SELECT d.name AS dog, c.name AS cat  
+FROM adoptions as ad
+FULL OUTER JOIN cats AS c ON ad.cat = c.id
+FULL OUTER JOIN dogs AS d ON ad.dog = d.id
+WHERE ad.dog IS NULL AND ad.cat IS NULL;
+
+SELECT v.name, d.name
```

Comment:
you using full outer join on all of them does not show me that you know when to use which kind of join ;)

Reply:
It's the only way to do it in a single query I think.